### PR TITLE
Cron and CI scripts refactoring

### PR DESCRIPTION
**Pull request by wjordan**
- refactor crontab from shell-invoked template to Chef [`cron`](https://docs.chef.io/resource_cron.html) resources
  - add [entry](https://github.com/code-dot-org/code-dot-org/pull/7705#discussion-diff-58802552R57) for `aws/ci_build`, so we can retire the `geoffrey@gateway` crontab
- refactoring CI build logic:
  - merge+rename: `build_and_mail_log`, `build` => `ci_build`
  - rename `build.rake` to `ci_build.rake`
  - `test-websites` now inherits duplicated build tasks from `websites`
  - add sensible `default` task to `ci_build.rake`

### CODE REVIEW

File: cookbooks/cdo-apps/recipes/crontab.rb [hunk 1]
```diff
@@ -1,17 +1,68 @@
...
+  cronjob m: '*/1',    cmd: "#{bin}/process_forms"
+  cronjob m: '*/1',    cmd: "#{bin}/deliver_poste_messages"
+  cronjob m: '*/1',    cmd: "#{bin}/geocode_hoc_activity"
+  cronjob m: 35,       cmd: "#{bin}/cron/analyze_hoc_activity"
+  cronjob h: 8, m: 55, cmd: "#{bin}/cron/ops_data_pull"
+  cronjob h: 5, m: 45, cmd: "#{bin}/cron/admin_stats"
+  cronjob h: 4, m: 40, cmd: "#{bin}/cron/funometer"
+  cronjob h: 6, m: 30, cmd: "#{bin}/cron/admin_progress"
+  cronjob h: 7, m: 10, cmd: "#{bin}/cron/retention_stats"
+  cronjob m: '*/1',    cmd: "#{root}/aws/ci_build", notify: 'dev+build@code.org'
```
Comment (wjordan) [inline cookbooks/cdo-apps/recipes/crontab.rb:57]: this added line should allow us to retire the `geoffrey@gateway` crontab.

**Comment by wjordan** [comment]
Since our CI system does not itself have an adequate testing setup (perhaps a good topic for another PR?), after this has been reviewed I'll just plan on hand-holding this PR through staging+test to work out any issues that may arise.

File: aws/ci_build [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+require 'cdo/hip_chat'
+require 'cdo/only_one'
+require 'mail'
+
+STARTED = 'build-started'
+
+def format_duration(total_seconds)
+  total_seconds = total_seconds.to_i
+  minutes = (total_seconds / 60).to_i
+  seconds = total_seconds - (minutes * 60)
```
Comment (mehalshah) [inline aws/ci_build:33]: total_seconds % 60

File: aws/ci_build [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    RakeUtils.git_fetch
+    count = RakeUtils.git_update_count
+    RakeUtils.git_pull if count > 0
+    count = [1, count].max
+
+    log = `git log --pretty=format:"%h %s (%an)" -n #{count}`
+    IO.write(deploy_dir('rebuild'), log)
+
+    HipChat.log "https://github.com/code-dot-org/code-dot-org/commit/#{`git rev-parse HEAD`}", message_format: 'text', color: 'purple'
```
Comment (mehalshah) [inline aws/ci_build:61]: Do we still need to HipChat?

File: aws/ci_build [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+def main()
+  start_time = Time.now.to_i
+  # Run the build command and exit if it didn't do anything.
+  status = 0
+  log = capture_stdout { status = build }
+  return status if status == 0 && log.empty?
+
+  Dir.chdir(deploy_dir) do
+    commit_url = "https://github.com/code-dot-org/code-dot-org/commit/#{RakeUtils.git_revision}"
+    # Notify the HipChat channels about what happened.
```
Comment (mehalshah) [inline aws/ci_build:94]: Update comment

**Comment by mehalshah** [comment]
Looks great - I look forward to retiring the old crontab. Let's talk over what the transition plan is for the old CI when you get a chance tomorrow

**Comment by wjordan** [comment]
looked good in my manual tests on test-kitchen w/ec2 driver. Will resolve conflicts and then merge (watching staging closely to make sure all goes well)

**Pull request merged**